Adds security vouchers and moves sabres into a side-grade from pistols

TsunamiAnt opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds Security Vouchers

Redeem them at the sec-tech for a either a pistol or a sword.

Allows carbon fiber sabre sheaths to be stored in suit storage slots.
Tweaks the logic of suit storage overlays a bit to make this possible.

ofc also removes both pistols and mags from default sec loadout, since you get them via voucher now.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

This brings the sabre from a decision of "sabres or stunweapons" to a decision of "sabres or pistols", which is overall a more reasonable choice to have the player make, because we still required them to carry nonlethal anyway, which means we present them with the illusion of a choice which in truth only has one correct answer. Obviously horrible design.

Vouchers are a better solution to this than say, beacons.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

Voucher and locker without sword, notice the gun is gone too.
<img width="805" height="1143" alt="image" src="https://github.com/user-attachments/assets/32b94473-e448-444b-b667-156ac4d1b863" />

Using the voucher
<img width="923" height="1024" alt="image" src="https://github.com/user-attachments/assets/dd2e82d7-3650-4d52-825b-d8b7fd5c7459" />

puts a box in your hand
<img width="702" height="256" alt="image" src="https://github.com/user-attachments/assets/20554ed7-6e12-4c55-bc02-282a402d343e" />

Sword on suit storage slot:
<img width="350" height="296" alt="image" src="https://github.com/user-attachments/assets/6124dd2d-fefd-44dd-891f-13a8b9876151" />

<img width="170" height="180" alt="image" src="https://github.com/user-attachments/assets/4d7f396d-3d8d-4d1a-aa17-16bf99cd6551" />

<img width="182" height="197" alt="image" src="https://github.com/user-attachments/assets/772500d3-355f-404c-bb79-8c3373ba616a" />

<img width="176" height="181" alt="image" src="https://github.com/user-attachments/assets/364ced14-c624-454f-ad18-2f3b7d6204ab" />

Sword on belt slot to illustrate how it's on the other side
<img width="188" height="177" alt="image" src="https://github.com/user-attachments/assets/2d12be85-549a-4f85-9625-53ba65db1864" />

</details>

## Changelog
:cl:
add: Added security vouchers, redeem at sec vendor
tweak: Moved sabres and pistols from loadouts and lockers into the voucher system.
balance: Made sabre sheaths fit in suit storages
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TsunamiAnt committed:
Adds suit storage compatibility to sabre sheath

Allows carbon fiber sabre sheaths to be stored in suit storage slots.

This is achieved by:
- Adding the ITEM_SLOT_SUITSTORE flag to the sheath.
- Updating the description to reflect the new functionality.
- Adding the sheath to the security_vest_allowed list.
- Fixing an issue where the suit storage overlay would not display correctly if item_state or icon_state were not defined.

PowerfulBacon reviewed (commented):
Does this render thr sprite on the opposite side of the player? If you have 2 swords at the same time, then you should have one on the left side of your body and the other on the right side.

TsunamiAnt committed:
Refactors suit store overlay handling

Moves the suit store overlay instantiation to a dedicated function.

Corrects pixel offset application for species-specific adjustments.
The X offset was previously adding instead of subtracting.

TsunamiAnt committed:
Adds security voucher redemption system.

Adds a voucher system to security vending machines.

This allows for the redemption of specific equipment through a menu-driven system.
Removes direct equipment spawning from job outfits, instead providing vouchers to redeem at vending machines.

TsunamiAnt committed:
Gives extra ammo to security vending machines

Provides an additional magazine to the security vending machine's NPS-10 loadout.

The mining DMI was also updated.

TsunamiAnt committed:
Adds deputy voucher box and carbon fiber sheath

Adds a security voucher box and spawns it in the warden's locker.

Adds carbon fiber sabres to the pistol spawner in the armory

Rukofamicom reviewed (approved):
I prefer this over the other one for two reasons:
* It makes more sense to force a choice between gun and saber than it does between belt and saber
* This PR doesn't show a graphical bug in the vending machine (stocking the sheath in machine shows it as empty)

Ultimately I like and approve both, but this is my choice to merge between the two

PowerfulBacon commented on code/modules/vending/security.dm:
```diff
@@ -49,3 +49,36 @@
+		if("NPS-10")
+			new /obj/item/gun/ballistic/automatic/pistol/security(drop_location)
+			new /obj/item/ammo_box/magazine/x200law(drop_location)
+			new /obj/item/ammo_box/magazine/x200law(drop_location)
```
Only give one spare clip of ammo, the single spare we have now already gives sec too much ammo for a green alert weapon, since they turn the pistol into a primary weapon that out-classes everything else in the armoury. The whole point of guns is that they are meant to be extremely powerful but expensive after all.

PowerfulBacon reviewed (commented):
I like this PR, but you need to fill out the 'why its good for the game section' properly.

TsunamiAnt committed:
Update security.dm

Tsar-Salat commented on code/modules/vending/security.dm:
```diff
@@ -49,3 +49,35 @@
+/obj/machinery/vending/security/proc/RedeemVoucher(obj/item/mining_voucher/security/voucher, mob/redeemer)
+	var/items = list("Carbon Fibre Sabre", "NPS-10")
+
+	var/selection = input(redeemer, "Pick your equipment", "Voucher Redemption") as null|anything in sort_list(items)
```
We are in 2025. Use tgui_inputs please.

PowerfulBacon commented on code/modules/vending/security.dm:
```diff
@@ -49,3 +49,35 @@
 	machine_name = "SecTech"
 	icon_state = "refill_sec"
+
+/obj/machinery/vending/security/proc/RedeemVoucher(obj/item/mining_voucher/security/voucher, mob/redeemer)
```
Also use snake_case for proc names

PowerfulBacon commented on code/modules/vending/security.dm:
```diff
@@ -49,3 +49,35 @@
+
+		if("NPS-10")
+			new /obj/item/gun/ballistic/automatic/pistol/security(drop_location)
+			new /obj/item/ammo_box/magazine/x200law(drop_location)
```
Put these items into the person's hands like normal vendor items do

TsunamiAnt committed:
Updates voucher redemption proc name

Renames the security voucher redemption proc for consistency.

This change ensures that the proc name follows the established naming conventions, improving code readability and maintainability.

TsunamiAnt committed:
Redeems security voucher into a gear box.

Changes the security voucher redemption process to spawn a box containing the selected equipment.

This ensures that the redeemed items are contained and easily managed, preventing them from being scattered around the vendor location. The spawned box is then given to the redeemer. If they can't hold it, it is dropped on the floor at their location.

TsunamiAnt committed:
Uses tgui for voucher selection

Updates the voucher redemption process to use `tgui_input_list` for a better user experience.

Renames local variable for clarity.

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into sheathstoarmor

TsunamiAnt committed:
Refactors suit storage icon handling.

Refactors how the suit storage overlay is generated and applied.

It now leverages the `build_worn_icon` proc for more consistent icon
generation and handles potential pixel offsets related to species DNA.
The layering of icons is also handled more consistently.

PowerfulBacon commented on code/modules/vending/security.dm:
```diff
@@ -49,3 +49,52 @@
+	if(!selection || !Adjacent(redeemer) || QDELETED(voucher) || voucher.loc != redeemer)
+		return
+
+	spawned_box = new /obj/item/storage/box/gearvend
```
Spawning a box just to insert one item into it is silly for the sword, additionally it also just gives you a box with a non-box sized item in it which allows for you to carry the sword/gun in a backpack.

TsunamiAnt committed:
Fixes voucher box oversight

TsunamiAnt committed:
Adds support for both boxed and single items

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into sheathstoarmor

Rukofamicom commented on code/modules/jobs/job_types/security_officer.dm:
```diff
@@ -162,16 +162,15 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
 	l_pocket = /obj/item/modular_computer/tablet/pda/preset/security
-	r_pocket = /obj/item/clothing/accessory/badge
-	suit_store = /obj/item/gun/ballistic/automatic/pistol/security
+	r_pocket = /obj/item/clothing/accessory/badge/officer
```
```suggestion
	r_pocket = /obj/item/clothing/accessory/badge
```

TsunamiAnt committed:
Update code/modules/jobs/job_types/security_officer.dm

Co-authored-by: Rukofamicom <oozelingcomplaints@gmail.com>

Tsar-Salat reviewed (approved):
I dont like the way the sprite on the suit overlay was done, but Im not going to block over it
